benthomasson/ansible-events-ui
# Aap 5457 lo log data

## Jira Ticket

[AAP-5457](https://issues.redhat.com/browse/AAP-5457)

## Description

This PR adds a new module `db.utils.lostream` for DB large object streaming. This will allow for reading and writing to the database, large blocks of text or binary data at arbitrary buffer sizes.

This module and its functionality are utilized in reading and writing ansible process log output from and to the database.

## Testing

0. Start with clean DB and run `alembic upgrade head`
1. Run server/ui.
2. Setup a hello-world job execution
3. Go to activation instance
4. You should see logs as usual.
5. Execute `psql`
6. `select log_id from activation_instance;` -- Note the log_id
7. `select * from pg_largeobject_metadata where oid = <log_id>;` -- you should see a matching record 
8. `select sum(length(data)) from pg_largeobject where loid = <log_id>;` -- will give length of the data
9. `select encode(lo_get(<log_id>, <length-of-data>), 'escape');` -- should show captured stdout
